cli/pkg/utils/api_client.go

```diff
@@ -31,7 +32,11 @@ func NewHTTPClient(conf *config.Backend) *client.KubeClarityAPIs {
 	if conf.DisableTLS {
 		transport = httptransport.New(conf.Host, client.DefaultBasePath, []string{"http"})
 	} else if conf.InsecureSkipVerify {
-		customTransport := http.DefaultTransport.(*http.Transport).Clone()
+		defaultTransport, ok := http.DefaultTransport.(*http.Transport)
+		if !ok {
```

Comment:
you should not continue here. and I think you can use a `nolint` comment here and not check for type assertion